### PR TITLE
Allows modifications to change environmental_resist. Adds nomex patching.

### PR DESCRIPTION
**Pull request by Petethegoat**

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -4255,6 +4255,9 @@ int item::get_env_resist( int override_base_resist ) const
     int resist_filter = get_var( "overwrite_env_resist", 0 );
     int resist = std::max( { resist_base, resist_filter, override_base_resist } );
 
+    resist += std::max(0, static_cast<int>(ceil(get_clothing_mod_val(
+        clothing_mod_type_environmental_resist))));
```
Comment (Petethegoat) [inline src/item.cpp:4259]: Should these lines be changes to follow your updated encumbrance example? Just a copy/paste job on my part.

Comment (lispcoc) [reply src/item.cpp:4259]: I think reading into original author of clothing mod feature, rounding up"Lower is better" value and rounding down "Higher is better" value is corrent.
so I think using `lround` is better in this case, like `bash_resist`.

**Comment by lispcoc** [comment]
CleverRaven#32399 is merged.

**Comment by Petethegoat** [comment]
Would you like me to update my changes here, or just PR them on the main repo?

**Comment by lispcoc** [comment]
I recommend making PR by yourself.

**Comment by Petethegoat** [comment]
Sounds good!

**Pull request closed without merging**